NirspecQuadFlatModel.__init__ fails when provided a NirspecFlatModel

The attempted call to `dynamic_mask` does not have the appropriate number of arguments:

https://github.com/spacetelescope/stdatamodels/blob/31d1199e6ca3ba3c57985c4ae4338f3140a5c563/src/stdatamodels/jwst/datamodels/nirspec_flat.py#L83

Attempts result in a failure:
```
>           self.quadrants[0].dq = dynamic_mask(self.quadrants[0])
E           TypeError: dynamic_mask() missing 1 required positional argument: 'mnemonic_map'
```

--- a/changes/401.bugfix.rst
+++ b/changes/401.bugfix.rst
@@ -1,0 +1,1 @@
+Fix bug preventing NirspecQuadFlatModel instantiation from NirspecFlatModel

--- a/src/stdatamodels/jwst/datamodels/nirspec_flat.py
+++ b/src/stdatamodels/jwst/datamodels/nirspec_flat.py
@@ -118,7 +118,7 @@ class NirspecQuadFlatModel(ReferenceFileModel):
             self.quadrants[0].wavelength = init.wavelength
             self.quadrants[0].flat_table = init.flat_table
             self.quadrants[0].dq_def = init.dq_def
-            self.quadrants[0].dq = dynamic_mask(self.quadrants[0])
+            self.quadrants[0].dq = dynamic_mask(self.quadrants[0], pixel)
             return
 
         super(NirspecQuadFlatModel, self).__init__(init=init, **kwargs)

--- a/src/stdatamodels/jwst/datamodels/tests/test_nirspec_flat.py
+++ b/src/stdatamodels/jwst/datamodels/tests/test_nirspec_flat.py
@@ -10,7 +10,14 @@ def flatmodel():
     rng = np.random.default_rng()
     data = rng.random((100, 100))
     dq = rng.integers(0, 2, (100, 100))
-    return NirspecFlatModel(data, dq=dq)
+    model = NirspecFlatModel(data, dq=dq)
+    model.meta.instrument.name = "NIRSPEC"
+    model.meta.reftype = "FLAT"
+    model.meta.author = "STSCI"
+    model.meta.description = "Test flat field model"
+    model.meta.pedigree = "GROUND"
+    model.meta.useafter = "2025-02-12T00:00:00"
+    return model
 
 
 def test_initialize_quad_from_flat(flatmodel):

--- a/src/stdatamodels/jwst/datamodels/tests/test_nirspec_flat.py
+++ b/src/stdatamodels/jwst/datamodels/tests/test_nirspec_flat.py
@@ -1,0 +1,23 @@
+"""Tests for nirspec flat field models."""
+
+import numpy as np
+import pytest
+from stdatamodels.jwst.datamodels import NirspecFlatModel, NirspecQuadFlatModel
+
+
+@pytest.fixture(scope="module")
+def flatmodel():
+    rng = np.random.default_rng()
+    data = rng.random((100, 100))
+    dq = rng.integers(0, 2, (100, 100))
+    return NirspecFlatModel(data, dq=dq)
+
+
+def test_initialize_quad_from_flat(flatmodel):
+    """
+    Cover a bug where attempting to initialize a quad flat model from a flat model would fail.
+    """
+    quadmodel = NirspecQuadFlatModel(flatmodel)
+    assert isinstance(quadmodel, NirspecQuadFlatModel)
+    assert np.allclose(quadmodel.quadrants[0].data, flatmodel.data)
+    assert np.allclose(quadmodel.quadrants[0].dq, flatmodel.dq)

